### PR TITLE
Bug Fix: Advanced Search - Fix query builder property search to search the right property

### PR DESCRIPTION
**Pull request by msnyder-msft**
### Summary of changes 🔍 
The query builder was always searching twin ids instead of the selected property.
Before
image

After
image

**Fixing duplicate values in dropdown**
Before

After
image

**Fixing query failing due to new line characters**
Before

After
image

### Testing 🧪
- [x]  select a string type property in advanced search and see the values in the auto complete are showing the values for that property

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

**Review by pasanchMSFT — APPROVED** [review]
The fix looks good to me, just curious about the behavior of the component with an empty string.

**Review by darsney — APPROVED** [review]

**Review by darsney — CHANGES_REQUESTED** [review]
Yeahh this just doesnt work, it always 400s

File: src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilderRow.tsx [hunk 1]
```diff
@@ -271,7 +268,7 @@ const QueryBuilderRow: React.FC<IQueryBuilderRowProps> = (props) => {
                     noOptionsText={t('advancedSearch.noOptionsFoundMessage')}
                     placeholderText={t('advancedSearch.valueFieldPlaceholder')}
                     resetInputOnBlur={false}
-                    searchPropertyName={DTID_PROPERTY_NAME}
+                    searchPropertyName={selectedProperty?.data.name || ''}
```
Comment (iremgokce) [inline src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilderRow.tsx:271]: is '' a valid search property name? should we keep DTID_PROPERTY_NAME in the right side of the ||?

**Review by iremgokce — APPROVED** [review]

**Review by darsney — APPROVED** [review]
LGTM

File: src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilder.tsx [hunk 2]
```diff
@@ -18,9 +18,12 @@ import {
 import QueryBuilderRow from './QueryBuilderRow';
 import { buildQuery } from './QueryBuilderUtils';
 import { useTranslation } from 'react-i18next';
-import { createGUID } from '../../../../Models/Services/Utils';
+import { createGUID, getDebugLogger } from '../../../../Models/Services/Utils';
 import { PropertyValueType } from '../../../../Models/Constants';
 
+const debugLogging = true;
```
Comment (pasanchMSFT) [inline src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilder.tsx:24]: remove logging

File: src/Components/TwinPropertySearchDropdown/TwinPropertySearchDropdown.tsx [hunk 3]
```diff
@@ -108,7 +109,7 @@ const TwinPropertySearchDropdown = (
     const searchTwinAdapterData = useAdapter({
         adapterMethod: (params: AdapterMethodParamsForSearchADTTwins) =>
             adapter.searchADTTwins(params),
-        refetchDependencies: [],
+        refetchDependencies: [searchPropertyName],
```
Comment (pasanchMSFT) [inline src/Components/TwinPropertySearchDropdown/TwinPropertySearchDropdown.tsx:112]: Why would you want to re-fetch here? This method is done on click.

File: src/Models/Services/Utils.ts [hunk 4]
```diff
@@ -483,6 +483,19 @@ export function sortAscendingOrDescending<T>(
...
 
+/**
+ * remove duplicate objects from an array
+ * @param array array of objects to operate on
+ * @param key key of the property to use as the discriminator
+ * @returns a new copy of the array with only unique values
+ */
+export function removeDuplicatesFromArray<T>(array: T[], key: keyof T) {
+    const check = new Set<string>();
+    return array.filter(
```
Comment (pasanchMSFT) [inline src/Models/Services/Utils.ts:494]: Why not `Array.from(check)`?

File: src/Components/TwinPropertySearchDropdown/TwinPropertySearchDropdown.tsx (hunk 3 above)
Comment (msnyder-msft) [reply src/Components/TwinPropertySearchDropdown/TwinPropertySearchDropdown.tsx:112]: this is in my twin property search. :)

File: src/Models/Services/Utils.ts (hunk 4 above)
Comment (msnyder-msft) [reply src/Models/Services/Utils.ts:494]: Short answer: I just copied a stack overflow post.
Long answer: I'm not sure you can have a set of objects. Or at least it doesn't help you with getting a distinct list. So instead, we just keep a list of distinct keys and only include items where the key is not found in the set already.

File: src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilder.tsx (hunk 2 above)
Comment (msnyder-msft) [reply src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilder.tsx:24]: good catch. Fixed

File: src/Models/Services/Utils.ts (hunk 4 above)
Comment (pasanchMSFT) [reply src/Models/Services/Utils.ts:494]: Fair enough, LGTM then

**Pull request merged**